app/src/main/java/com/elmeradrianv/shesafe/SignUpActivity.java

```diff
@@ -0,0 +1,177 @@
+        EditText etFirstName = findViewById(R.id.etFirstName);
+        EditText etLastName = findViewById(R.id.etLastName);
+        EditText etEmail = findViewById(R.id.etEmail);
+        EditText etPersonalDescription = findViewById(R.id.etPersonalDescription);
+        EditText etPassword = findViewById(R.id.etPassword);
+        EditText etPasswordConfirm = findViewById(R.id.etPasswordConfirm);
+        Button btnSignUp = findViewById(R.id.btnSignUp);
+        Button btnAddProfilePhoto = findViewById(R.id.btnAddProfilePhoto);
+
+
+        setImage("", ivProfilePhoto);
+        setupBtnSignUp(btnSignUp, etUsername, etFirstName, etLastName, etEmail, etPersonalDescription, etPassword, etPasswordConfirm);
+        setBtnAddProfilePhoto(btnAddProfilePhoto);
+
+
+    }
+
+
+    private void setupBtnSignUp(Button btnSignUp, EditText etUsername, EditText etFirstName, EditText etLastName, EditText etEmail, EditText etPersonalDescription, EditText etPassword, EditText etPasswordConfirm) {
+        btnSignUp.setOnClickListener(v -> {
+            String username = etUsername.getText().toString();
+            String firstName = etFirstName.getText().toString();
+            String lastName = etLastName.getText().toString();
+            String email = etEmail.getText().toString();
+            String personalDescription = etPersonalDescription.getText().toString();
+            String password = etPassword.getText().toString();
+            String passwordConfirm = etPasswordConfirm.getText().toString();
+            if (username.isEmpty() && firstName.isEmpty() && lastName.isEmpty() && email.isEmpty() && personalDescription.isEmpty() && password.isEmpty() && passwordConfirm.isEmpty()) {
+                Toast.makeText(this, "Please fill all the fields", Toast.LENGTH_SHORT).show();
+            } else {
+                if (password.equals(passwordConfirm)) {
+                    signupNewUser(username, firstName, lastName, email, personalDescription, password);
+                } else {
+                    Toast.makeText(this, "Password doesn't matched", Toast.LENGTH_SHORT).show();
+                }
+            }
+
+        });
+    }
+
+
+    private void signupNewUser(String username, String firstName, String lastName, String email, String personalDescription, String password) {
+        ParseUser user = new ParseUser();
+        user.setUsername(username);
+        user.setEmail(email);
+        user.setPassword(password);
+        user.put(User.PERSONAL_DESCRIPTION_KEY, personalDescription);
+        user.put(User.FIRST_NAME_KEY, firstName);
+        user.put(User.LAST_NAME_KEY, lastName);
+        user.signUpInBackground(e2 -> {
+            if (e2 != null) {
+                Toast.makeText(this, "Couldn't sign up", Toast.LENGTH_SHORT).show();
+                Log.e(TAG, "signupNewUser: Signup error", e2);
+                return;
+            }
+            if (PERSONALIZED_PHOTO_PICKED) {
```

Comment:
rather than checking "Personalized_photo_picked" can you just check if profilePhoto is null?